src/Bicep.LangServer.UnitTests/Handlers/BicepDecompileCommandHandlerTests.cs

```diff
@@ -0,0 +1,752 @@
+  name: 'storageaccount'
+  tags: {
+    displayName: 'storageaccount'
+  }
+  location: resourceGroup().location
+  kind: 'StorageV2'
+  sku: {
+    name: 'Premium_LRS'
+    tier: 'Premium'
+  }
+}
+
+module nestedDeploymentInner2 './nested_nestedDeploymentInner2.bicep' = {
+  name: 'nestedDeploymentInner2'
+  params: {
+  }
+}";
+
+        private const string ComplexExpectedBicep_Filename2 = "nested_nestedDeploymentInner.bicep";
+        private const string ComplexExpectedBicep_Output2Regex = @"resource storageaccount1 'Microsoft.Storage/storageAccounts@2021-04-01'";
+
+        private const string ComplexExpectedBicep_Filename3 = "nested_nestedDeploymentInner2.bicep";
+        private const string ComplexExpectedBicep_Output3Regex = "^$";
+
+        private const string ComplexExpectedBicep_Filename4 = "nested_nestedDeploymentOuter.bicep";
+        private const string ComplexExpectedBicep_Output4Regex = "resource storageaccount2 'Microsoft.Storage/storageAccounts@2021-04-01'";
+
+        #endregion
+
+        private (BicepDecompileCommandHandler, BicepDecompileSaveCommandHandler) CreateCommandHandlers(LanguageServerMock server)
+        {
+            return (
+                new BicepDecompileCommandHandler(
+                    Repository.Create<ISerializer>().Object,
+                    BicepTestConstants.FeatureProviderFactory,
+                    BicepTestConstants.NamespaceProvider,
+                    BicepTestConstants.RegistryProvider,
+                    server.Mock.Object,
+                    BicepTestConstants.CreateMockTelemetryProvider().Object,
+                    BicepTestConstants.ApiVersionProviderFactory,
+                    BicepTestConstants.LinterAnalyzer,
+                    BicepTestConstants.FileResolver),
+                new BicepDecompileSaveCommandHandler(
+                    Repository.Create<ISerializer>().Object,
+                    BicepTestConstants.FeatureProviderFactory,
+                    BicepTestConstants.NamespaceProvider,
+                    BicepTestConstants.RegistryProvider,
+                    server.Mock.Object,
+                    BicepTestConstants.CreateMockTelemetryProvider().Object,
+                    BicepTestConstants.ApiVersionProviderFactory,
+                    BicepTestConstants.LinterAnalyzer,
+                    BicepTestConstants.FileResolver,
+                    server.ClientCapabilitiesProvider,
+                    server.Mock.Object)
+                );
+        }
+
+        [DataRow(null)]
+        [DataRow("")]
+        [DataRow("   ")]
+        [DataTestMethod]
+        public void WithInvalidPath_ShouldThrowArgumentException(string path)
+        {
+            var server = new LanguageServerMock();
+
+            ICompilationManager bicepCompilationManager = Repository.Create<ICompilationManager>().Object;
+            var (handler, _) = CreateCommandHandlers(server);
+
+#pragma warning disable VSTHRD002 // Throws synchronously
```

Comment:
Can we change this to below instead of using #pragma statements?
```
Action action = () => handler.Handle(new(DocumentUri.File(path)), CancellationToken.None);
action.Should().Throw<Exception>();
```